classes/Validate.php

```diff
@@ -1258,4 +1261,27 @@ public static function isValidImapUrl($imapUrl)
 
         return true;
     }
+
+    /**
+     * Search for dot positions in given string
+     *
+     * @param string $string
+     * @param string $dotCharacter
+     *
+     * @return array offsets of dots in given string
+     */
+    protected static function findAllDotOffsets($string, $dotCharacter)
+    {
+        $matches = array();
+
+        preg_match_all('#' . $dotCharacter . '#', $string, $matches, PREG_OFFSET_CAPTURE);
```

Comment:
To avoid testing a space later, have you tried something like

```suggestion
        preg_match_all('#' . $dotCharacter . '(?![[:space:]])#', $string, $matches, PREG_OFFSET_CAPTURE);
```